### ISSUE TITLE
[Suggestion]: Rework the Withered Symbiote's arrow barrage

### ISSUE THREAD

**Issue by GearsawStudios** [nonamecrackers2/crackers-wither-storm-mod#1652]
### Description of feature request

If you've ever fought the Withered Symbiote, one attack stands out as highly underwhelming, that being the arrow barrage. Although visually it's an amazing attack, combat wise it's a weaker version of the fireball halo and wither skull barrage attacks. It adds nothing to the fight besides showing off the Symbiote's powers. Even worse, it can only land one arrow at a time, which massively degrades the quality of the attack. Thirty or so arrows, twenty-nine bounce off and one hits, how disappointing. How would one go about improving it then? I'd argue that starting from the ground up would work best.

To understand what attack should its purpose be, one must think of what attacks the Symbiote currently has and what niche it fills.

Fireball Halo is a block destroying attack that works from any range that disturbs the battlefield.

Wither Skull Barrage is a highly damaging (and overtuned but that's a separate problem) attack that forces the player away and essentially does the same thing as the fireballs.

Arrow Barrage serves practically the same niche as the above two without destroying blocks and with far lower success rate. The only new thing it brings to the table is that it has a large projectile size and ability to utterly destroy shields (which is done well enough by the other barrage attacks anyways.)

The Pull forces the player near the symbiont and sets them up to be destroyed by the melee attack while forcing the player off any towers.

The Melee is a general melee attack that usually kills the player instantly with iron armor (of which I dislike) that prevents running through the Symbiont and prevents invalidating the summoning phases.

The Slam works as a perfect mobility tool for the Symbiont, preventing escapes and most cheeses.

The summoning needs no explanation as it sets up the core of the fight and differentiates it from most other fights.

The evoker fangs are used to prevent underground cheesing, but often isn't used often enough to truly do anything about it.

Knowing these, what doesn't the Symbiont have in its move set? It doesn't have a reliable way to hit far away targets without repositioning with a slam. To improve the attack, it should move its focus from a mid range to more of a long range attack. Of course, implementation should be done as seen fit. However, I have some concepts that may work and potentially might be considered for implementation separately.

1. Raise the attack's firing point twenty blocks or so, which massively increases the spread and therefore area it covers. Not only does this build upon the attacks main gimmick (projectile size,) it expands its niche to crowd control. Essentially, it turns into an arrow rain that forces the player(s) from an area by using an explosion like attack. If the bursts are merged together, then it could have each one fire at different intervals, hitting multiple times to balance how easy it would be to dodge.
![image](https://github.com/nonamecrackers2/crackers-wither-storm-mod/assets/122060314/215f38c5-7b14-47c1-89fc-2b9a4f0363d1) Proof of concept for attack 1 (My favorite)

2. Change the arrows to spawn above a random player and fall down over the course of a few seconds, forcing the player to find cover, water or continue running for the duration of the attack. While less interesting, it'd work much better than the existing attack. 
![image](https://github.com/nonamecrackers2/crackers-wither-storm-mod/assets/122060314/6d3f5543-b006-483c-907f-441136e3984b) Proof of concept for attack 2

3. Change the barrage into a volcano of arrows spawning above the symbiont that spans roughly ten or so blocks around the symbiont with standard distribution. This should have a long charge up to prevent cheap hits. While fulfilling a completely different idea, it'd work wonders if combined with the pull attack in killing the player, as players would either have to shield the attack, sprint or fall into a nearby hole. Potentially, add a safe zone around the symbiont itself to make it a risky way to dodge the attack due to the potential of a melee follow up. 
![image](https://github.com/nonamecrackers2/crackers-wither-storm-mod/assets/122060314/3dcb42cc-03f9-422e-86fd-5bde5a91afb9) Proof of concept for attack 3

Simply put, the attack isn't great where it stands due to how repetitive it is and desperately needs a rework to give it a real use in the fight besides a watered down version of other attacks. Hopefully these concepts provide a means of improving the attack outside of being visually amazing.

**Comment by Laikagor** [comment]
you wrote a whole essay on this too huh

**Comment by GearsawStudios** [comment]
> you wrote a whole essay on this too huh

I will write more essays in the near future.

**Comment by Aimyx** [comment]
> I will write more essays in the near future.

epic

**Comment by watersofnazareth** [comment]
Completed, I forgor about this issue lol
